### PR TITLE
Update flatMap documentation for Interpreter V2/Ergo 5.0

### PR DESCRIPTION
**Pull request by SethDusek**
These rules don't apply to newer versions of interpreter

### CODE REVIEW

**Review by kushti — APPROVED** [review]

File: docs/spec/generated/predeftypes.tex [hunk 1]
```diff
@@ -1,27 +1,31 @@
...
+\lst{Unit}	&	$98$	&	\lst{isConst}	& \lst{true}	&	\lst{false} &	\lst{false}	& Sec.~\ref{sec:type:Unit} \\
 \hline
-\lst{Context}	&	$101$	&	\lst{false}	& \lst{false}	&	\lst{false} &	\lst{false}	& Sec.~\ref{sec:type:Context} \\
+\lst{Box}	&	$99$	&	\lst{isConst}	& \lst{false}	&	\lst{false} &	\lst{false}	& Sec.~\ref{sec:type:Box} \\
 \hline
-\lst{Header}	&	$104$	&	\lst{true}	& \lst{false}	&	\lst{false} &	\lst{false}	& Sec.~\ref{sec:type:Header} \\
+\lst{AvlTree}	&	$100$	&	\lst{isConst}	& \lst{false}	&	\lst{false} &	\lst{false}	& Sec.~\ref{sec:type:AvlTree} \\
 \hline
-\lst{PreHeader}	&	$105$	&	\lst{true}	& \lst{false}	&	\lst{false} &	\lst{false}	& Sec.~\ref{sec:type:PreHeader} \\
+\lst{Context}	&	$101$	&	\lst{isConst}	& \lst{false}	&	\lst{false} &	\lst{false}	& Sec.~\ref{sec:type:Context} \\
```
Comment (aslesarenko) [inline docs/spec/generated/predeftypes.tex:25]: @SethDusek Why this change to `\lst{isConst}` is necessary?

Comment (SethDusek) [reply docs/spec/generated/predeftypes.tex:25]: This change is from the auto-generated documentation (GenPredefTypesApp.scala). Since the .tex files weren't updated in 3 years there are some changes like AvlTree now having documentation for methods

**Comment by aslesarenko** [comment]
@SethDusek, please resolve conflicts. 
Also add the following line to the `Publish JS snapshot ${{ github.ref }}` task of `ci.yml`.
```if: env.HAS_SECRETS == 'true'```

[Example](https://github.com/ScorexFoundation/sigmastate-interpreter/blob/71e67c4dbbabfad1f732f1244d251c815fe80474/.github/workflows/ci.yml#L109)

**Review by aslesarenko — APPROVED** [review]
